### PR TITLE
getByBlockNumber replaced by getByBlockHeader on tests

### LINKED ISSUE #5165: Fix tests to getByBlockHeader or create a test ProtocolSchedule that has getByBlockNumber inc streamMilestoneBlocks for checking milestones
Part of #4789

### PR DESCRIPTION
**Pull request by gfukushima**
## PR description
This PR removes the getByBlockNumber from tests and replace it's usage by getByBlockHeader.

## Fixed Issue(s)
Fixes #5165

### CODE REVIEW

File: consensus/merge/src/test/java/org/hyperledger/besu/consensus/merge/MergeProtocolScheduleTest.java [hunk 1]
```diff
@@ -27,6 +30,8 @@
 
 public class MergeProtocolScheduleTest {
 
+  final BlockDataGenerator blockDataGenerator = new BlockDataGenerator();
```
Comment (siladu) [inline consensus/merge/src/test/java/org/hyperledger/besu/consensus/merge/MergeProtocolScheduleTest.java:33]: redundant now?

**Review by siladu — APPROVED** [review]
LGTM

File: consensus/merge/src/test/java/org/hyperledger/besu/consensus/merge/MergeProtocolScheduleTest.java (hunk 1 above)
Comment (gfukushima) [reply consensus/merge/src/test/java/org/hyperledger/besu/consensus/merge/MergeProtocolScheduleTest.java:33]: Removed Thanks!

File: besu/src/test/java/org/hyperledger/besu/PrivacyTest.java [hunk 2]
```diff
@@ -145,8 +147,12 @@ private PrecompiledContract getPrecompile(
       final BesuController besuController, final Address defaultPrivacy) {
     return besuController
         .getProtocolSchedule()
-        .getByBlockNumber(1)
+        .getByBlockHeader(blockHeader(0))
```
[the remark this answers is not in the document]
Comment (gfukushima) [reply besu/src/test/java/org/hyperledger/besu/PrivacyTest.java:150]: Replicating the production code scenario

[the remark this answers is not in the document]
Comment (jframe) [reply besu/src/test/java/org/hyperledger/besu/PrivacyTest.java:150]: I remember that change to use the genesis block now. Make sense.

**Pull request merged**